src/server.cpp

```diff
@@ -411,10 +411,20 @@ CServer::CServer ( const int          iNewMaxNumChan,
         vecChannels[i].SetEnable ( true );
     }
 
-    // introduced by kraney (#653): "increased the size of the thread pool"
-    if ( bUseMultithreading )
+    int iAvailableCores = QThread::idealThreadCount();
+
+    // setup QThreadPool if multithreading is active and possible
+    if ( bUseMultithreading && iAvailableCores > 1 )
     {
-        QThreadPool::globalInstance()->setMaxThreadCount ( QThread::idealThreadCount() * 4 );
+        // set maximum thread count to available -1 to keep one thread for the main event loop
+        iMaxNumThreads = iAvailableCores - 1;
```

Comment:
The ` - 1` may vary.  The GUI adds a thread, as does the recorder.  Whilst neither of those is real time process priority as the audio threads should be, they may need to be taken into account.

Reply:
I think one core is enough for gui, recording, and such. 
I did more tests with up to 80 clients and it works quite nicely, without gui though in my case.